src/pharmpy/tools/iivsearch/algorithms.py

```diff
@@ -149,3 +153,28 @@ def update_description(model):
     description = create_description(model)
     model = model.replace(description=description)
     return model
+
+
+def _get_eta_from_parameter(model: Model, parameters: List[str]) -> List[str]:
+    # returns list of eta parameters from parameter names
+    iiv_set = set()
+    iiv_names = model.random_variables.names
+    if _has_iiv(model, parameters):
```

Comment:
Is it needed to check if it has IIV? If `param` is an empty list it means that there is no IIV on that parameter?

Reply:
If you want the error check that is in `_add_iiv` you can have that when validating input: https://github.com/pharmpy/pharmpy/blob/f357a01c2242be765dc93ddd7b94358d153f88fd/src/pharmpy/tools/iivsearch/tool.py#L259-L280